### PR TITLE
added new tests in agent_test.go

### PR DESCRIPTION
**Pull request by h0x0er**
#### added test functions for:
* writeDone()
* writeStatus()

#### added a new test_case in TestRun function that triggers a specific block of code in agent.go; requiring the command arg to be nil.

agent_test_

### CODE REVIEW

**Review by varunsh-coder — CHANGES_REQUESTED** [review]

**Comment by varunsh-coder** [comment]
@h0x0er I think we discussed this offline. Please let me know if anything pending on my side.

File: agent.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func Run(ctx context.Context, configFilePath string, hostDNSServer DNSServer,
 
 	config := &config{}
 	if err := config.init(configFilePath); err != nil {
-		writeStatus(fmt.Sprintf("Error reading config file %v", err))
+		_ = writeStatus(fmt.Sprintf("Error reading config file %v", err))
```
Comment (varunsh-coder) [inline agent.go:62]: @h0x0er you can remove the `_ =` part. That is redundant. Same applies to other places as well.

**Comment by varunsh-coder** [comment]
@h0x0er once you make the changes, please create a PR into `rcbranch`. I will then merge changes into `int` to do integration testing.

**Comment by h0x0er** [comment]
@varunsh-coder I am closing this PR and creating a new one into `rcbranch`

**Pull request closed without merging**